studio/pages/project/[ref]/functions/[functionSlug]/index.tsx

```diff
@@ -135,28 +222,140 @@ const PageLayout: NextPageWithLayout = () => {
+                    customDateFormat={selectedInterval.format}
+                    stackColors={['brand', 'slate', 'yellow', 'red']}
+                    onBarClick={() => {
+                      router.push(
+                        `/project/${projectRef}/functions/${functionSlug}/invocations?its=${startDate.toISOString()}`
+                      )
+                    }}
+                  />
+                )
+              }}
+            />
+          ) : (
+            <ReportWidget
+              title="Invocations"
+              data={invocationsChartData}
+              isLoading={isInvStatsChartLoading}
+              renderer={(props) => (
+                <BarChart
+                  className="w-full"
+                  xAxisKey="timestamp"
+                  yAxisKey="count"
+                  data={props.data}
+                  highlightedValue={sumBy(props.data, 'count')}
+                  customDateFormat={selectedInterval.format}
+                  onBarClick={(v) => {
+                    router.push(
+                      `/project/${projectRef}/functions/${functionSlug}/invocations?its=${startDate.toISOString()}&ite=${
+                        v.timestamp
+                      }`
+                    )
+                  }}
+                />
+              )}
+            />
+          )}
+          {resourceUsageMetricsEnabled && (
+            <>
+              <ReportWidget
+                title="CPU time"
+                tooltip="Average CPU time usage for the function"
+                data={resourceChartData}
+                isLoading={isResourceUsageChartLoading}
+                renderer={(props) => {
+                  const latest = normalizedResourceUsageData[normalizedResourceUsageData.length - 1]
+                  let highlightedValue
+                  if (latest) {
+                    highlightedValue = latest['avg_cpu_time_used']
+                  }
+                  return (
+                    <AreaChart
+                      className="w-full"
+                      xAxisKey="timestamp"
+                      customDateFormat={selectedInterval.format}
+                      yAxisKey="avg_cpu_time_used"
+                      data={props.data}
+                      format="ms"
+                      highlightedValue={highlightedValue}
+                    />
                   )
                 }}
               />
-            )}
-          />
+              <ReportWidget
+                title="Memory"
+                tooltip="Average memory usage for the function"
+                data={resourceChartData}
+                isLoading={isResourceUsageChartLoading}
+                renderer={(props) => {
+                  const latest = normalizedResourceUsageData[normalizedResourceUsageData.length - 1]
+                  let highlightedValue
+                  if (latest) {
+                    highlightedValue = latest['avg_memory_used']
```

Comment:
might want to handle precision formatting here (or in the sql query is fine as well). there are some precision util functions already available.

Reply:
Number formatting is done within the StackedBarChart component.

Reply:
I was referring to the highlighted value on hover, i think StackedBarChart doesn't have the `valuePrecision` prop yet. Something to consider if you dynamic precisions like in the ingress/egress charts

https://github.com/supabase/supabase/blob/b91532f6c7598f3b6127deae323e2991658faaa3/studio/components/interfaces/Reports/renderers/ApiRenderers.tsx#L34

Reply:
Ah I see what you mean. I guess the default is fine for now here.

I added `valuePrecision` prop to StackedBarChart in this PR. It's used here for numberFormatter https://github.com/supabase/supabase/blob/3770de89b8ed42b8f475d8cf0de351397e2096f0/apps/studio/components/ui/Charts/StackedBarChart.tsx#L90